### PR TITLE
Add vRequestStatus

### PR DESCRIPTION
**Pull request by niccokunzmann**
## Linked issue

- Contributes to #1135 

## Description

xCal requires special parsing and creation of REQUEST-STATUS

This adds a vText subclass, so that everything works as before.

## Checklist

- [x] I added a change log entry, following the instructions in all subsections under [Change log requirements](https://icalendar.readthedocs.io/en/latest/contribute/#change-log-requirements).
- [x] I followed icalendar's [Artificial intelligence policy](https://icalendar.readthedocs.io/en/latest/contribute/index.html#artificial-intelligence-policy) and disclosed my [Responsible AI use](https://icalendar.readthedocs.io/en/latest/contribute/index.html#responsible-ai-use) in my commit messages, if applicable.
- [x] I added or updated tests, if applicable.
- [x] I ran and ensured all tests pass locally by following [Run tests](https://icalendar.readthedocs.io/en/latest/contribute/development.html#run-tests).
- [x] I added or edited documentation as necessary, both as docstrings to be rendered in the API documentation and narrative documentation, following the [Style guide](https://icalendar.readthedocs.io/en/latest/contribute/documentation/style-guide.html).

### CODE REVIEW

File: src/icalendar/prop/request_status.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+        Parameters:
+            jcal_property: The jCal property to parse.
+
+        Raises:
+            ~error.JCalParsingError: If the provided jCal is invalid.
+        """
+        JCalParsingError.validate_property(jcal_property, cls)
+        JCalParsingError.validate_list_type(jcal_property[3], str, cls, 3)
+        return cls.new(
```
Comment (josemmiranda04) [inline src/icalendar/prop/request_status.py:166]: I found one malformed-jCal edge case here.

If `jcal_property[3]` contains more than 3 items, `cls.new(*jcal_property[3], params=...)` raises:

`TypeError: vRequestStatus.new() got multiple values for argument 'params'`

File: src/icalendar/tests/prop/test_request_status.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    assert request_status.data is None
+    assert request_status == "3.4;"
+
+
+@mark_examples
+def test_from_jcal(text, code, description, data, text_serialized):
+    """Check that jcal round trip preserves semicolon escape."""
+    request_status = vRequestStatus.new(code, description, data)
+    jcal = request_status.to_jcal("request-status")
+    print(jcal)
```
Comment (josemmiranda04) [inline src/icalendar/tests/prop/test_request_status.py:128]: Small suggestion: there is a `print(jcal)` left in `test_from_jcal()`. I think it can be removed to keep the test output clean.

**Review by josemmiranda04 — COMMENTED** [review]
I reviewed the `vRequestStatus` implementation and tests locally. The targeted tests pass, and I found one malformed jCal edge case plus a small debug print that i think it could be changed.

**Comment by niccokunzmann** [comment]
@josemmiranda04 Thank you for the unexpected review! Very useful. I think, your skills can benefit the project. I welcome your review on other PRs, too.

I added tests for your last review. Please have a look.

**Comment by josemmiranda04** [comment]
> @josemmiranda04 Thank you for the unexpected review! Very useful. I think, your skills can benefit the project. I welcome your review on other PRs, too.
> 
> I added tests for your last review. Please have a look.

Thanks for the update. I reviewed the new edge-case tests and the from_jcal() changes locally. The jCal case is handled correctly, and all test_request_status.py tests pass (54). The debug print is also gone. Looks good to me.
And thank you for the feedback! I will try to be useful on other PRs :)

**Comment by niccokunzmann** [comment]
Thanks for the review again! I will wait for a docs review now before merging.

**Comment by josemmiranda04** [comment]
> Thanks for the review again! I will wait for a docs review now before merging.

I can take a look at the docs review tomorrow!

File: src/icalendar/prop/request_status.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        if self.__match is None:
+            self.__match = REQUEST_STATUS_GRAMMAR.match(self)
+        return self.__match
+
+    @property
+    def code(self) -> tuple[int, ...]:
+        """Return the status code as a tuple.
+
+        Returns:
+            A tuple of integers or () if the status code could not be parsed.
```
Comment (josemmiranda04) [inline src/icalendar/prop/request_status.py]: The `Returns` description seems to have been copied from `code`. `code_string` returns a string (for example "2.0") or "" when it cannot be parsed, rather than a tuple of integers.

File: src/icalendar/prop/request_status.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+            return None
+        data: str = match.group("data")
+        if data is None:
+            return None
+        return self._unescape(data)
+
+    @classmethod
+    def new(
+        cls,
+        code: Sequence[int, ...] | str | int,
```
Comment (josemmiranda04) [inline src/icalendar/prop/request_status.py]: The generated API docs currently render this as `Sequence[int, Ellipsis]`. Since `Sequence` takes a single item type, should this be `Sequence[int] | str | int` instead?

**Review by josemmiranda04 — COMMENTED** [review]
I reviewed the generated API docs for `vRequestStatus`. The page renders correctly in Sphinx, but I found two docs/type-hint issues that i think affects the generated API reference.

**Comment by niccokunzmann** [comment]
@josemmiranda04 Thanks for catching this. I updated the typing and the docs.
